envi/__init__.py

```diff
@@ -108,6 +96,47 @@
     'mips64':       ARCH_MIPS64,
 }
 
+arch_aliases = {
+    'i386':             ('i486', 'i586', 'i686', 'x86'),
+    'amd64':            ('x86_64',),
+    'arm':              ('armv6l', 'armv7l', 'a32'),
+    'thumb':            ('t32', 'thumb2'),
+    'a64':              ('aarch64',),
+    'ppc32-embedded':   ('ppc32',),
+    'ppc-embedded':     ('ppc64-embedded','ppc-spe'),
+    'ppc-server':       ('ppc64-server','altivec', 'ppc-altivec'),
+    'ppc-vle':          ('vle','ppc32-vle', 'ppcvle'),
+    'rv32':             ('riscv', 'risc-v',),
+    'mcs51':            ('8051', '80x51'),
+    'rxv2':             ('rxv1', 'rx'),
+    'mips32':           ('mips',),
+}
+
+arch_module_path = {
```

Comment:
I feel like if we tweak this to something like a a dictionary of dictionaries, we could collapse arch_aliases and arch_module_path into a more coherent unified structure that doesn't require looping to find.

Also, I understand the necessity, but it strikes me a bit kludgy that we need to encode the name of the thing we're importing here. Ideally, this module (or some helper or something), would discover the name automagically and we wouldn't need to encode it twice. Now that might be a pipe dream, but I think it'd be good to discuss.

Reply:
i'm open to the discussion.  it's probably time that we collapse things into a more cohesive storage mechanism.
this PR or another (post-release, which is so long overdue)?

Reply:
Another. It feels like a separate conversation.